### PR TITLE
Chore: Remove @eslint/compat and don't run all eslint rules in betterer

### PR DESCRIPTION
**Pull request by tomratcliffe**
**What is this feature?**
Ensures that betterer doesn't run eslint rules that are checked anyway in separate linting steps

**Why do we need this feature?**
Brings betterer check time back to what it used to be, ~1min (rather than ~1:30).
Also means that the betterer step will only fail for betterer issues, rather than linting issues that would be caught in the eslint step anyway

Locally, I've seen it roughly halve the time for running `yarn betterer -u` against all files

**Who is this feature for?**
UI devs

### CODE REVIEW

File: .betterer.eslint.config.js
Comment (tomratcliffe) [inline .betterer.eslint.config.js:1]: I would rather import this from the main eslint config, but I would end up causing a circular dependency, so I duplicated it slightly.

The main reasoning here being to make the betterer config work independently to the main one, so it won't try and run the same linting twice

File: .betterer.ts [hunk 1]
```diff
@@ -75,17 +72,14 @@ function regexp(pattern: RegExp, issueMessage: string) {
...
+  return new BettererFileTest(async (filePaths, fileTestResult) => {
     // Just bail early if there's no files to test. Prevents trying to get the base config from failing
     if (filePaths.length === 0) {
       return;
     }
 
-    const { baseDirectory } = resolver;
-
     const runner = new ESLint({
-      overrideConfig: config,
```
Comment (tomratcliffe) [inline .betterer.ts:87]: Overriding this way seems to mean that it doesn't extend the config, and instead uses the betterer config in isolation

File: eslint.config.js [hunk 2]
```diff
@@ -1,6 +1,5 @@
 // @ts-check
 const emotionPlugin = require('@emotion/eslint-plugin');
-const { fixupPluginRules } = require('@eslint/compat');
```
Comment (tomratcliffe) [inline eslint.config.js:3]: The testing-library plugin was [updated to 6.4.0 recently](https://github.com/grafana/grafana/pull/96391) which includes the fixed config, so we don't need to run it through compat anymore

**Review by jackw — APPROVED** [review]
Awesome! 🚀

**Pull request merged**